var/spack/repos/builtin/packages/openpmd-api/package.py

```diff
@@ -47,17 +47,14 @@ class OpenpmdApi(CMakePackage):
     #         description='Enable JSON support')
     variant('python', default=True,
             description='Enable Python bindings')
-    variant('test', default=True,
-            description='Build the tests')
 
     depends_on('cmake@3.10.0:', type='build')
-    depends_on('boost@1.62.0:')
     depends_on('mpark-variant@1.3.0:')
-    depends_on('catch@2.2.1: ~single_header', when='+test', type='build')
+    depends_on('catch@2.2.1: ~single_header', type='test')
```

Comment:
@adamjstewart as soon as I change the type from `build` to `test` here, `spack` aborts with:
```bash
$ spack install --test=root openpmd-api
==> Error: There are no valid versions for spectrum-mpi that match ':'
```

Reply:
Oh, I see this all the time, just rerun the command a couple of times.

Reply:
Tried that but it's very persistent with me :)

Reply:
Has anyone tried adding a fake version of `spectrum-mpi`? I've never ran into this bug myself. I assume you aren't using `spectrum-mpi`, right?

Reply:
For the record, `spectrum-mpi` is the _only_ package in Spack without a single version, so if it this was causing some weird concretization bug I wouldn't be that surprised.

Reply:
Yep, I am not depending on nor have I installed `spectrum-mpi`. But I do have a mpi dependency.

Removing my mpi dependency also circumvents the concretization issue:
```
$ spack install --test=root openpmd-api ~mpi
# ok
```

Reply:
Migrated issue to #8398

Reply:
problem gone with latest `develop`